File: threading_pub_sub/threading_pub_sub/src/main/java/helpers/FieldGenerator.java
package helpers;

import java.math.BigDecimal;
import java.math.RoundingMode;
import java.time.LocalDate;
import java.util.Arrays;
import java.util.List;
import java.util.Random;

public class FieldGenerator {
	protected static final List<String> companies = Arrays.asList("Google", "Bing", "Yahoo", "Fluentis", "Mind",
			"Amazon", "Conduent", "Continental", "Bitdefender");
	protected static final List<LocalDate> dates = Arrays.asList(LocalDate.of(2022, 3, 30), LocalDate.of(2021, 2, 11),
			LocalDate.of(2022, 4, 1), LocalDate.of(2011, 4, 20), LocalDate.of(2000, 1, 1));
	protected static final List<String> operators = Arrays.asList("=", ">", "<", ">=", "<=", "!=");
	public static final Random rng = new Random();
	public static final int RANGE_MIN = 90;
	public static final int RANGE_MAX = 99;

	private FieldGenerator() {
	}

	public static double getRandomDouble() {
		double value = RANGE_MIN + (RANGE_MAX - RANGE_MIN) * rng.nextDouble();
		BigDecimal bd = BigDecimal.valueOf(value);
		bd = bd.setScale(1, RoundingMode.HALF_UP); // 1 cifra dupa virgula
		return bd.doubleValue();
//		return RANGE_MIN + (RANGE_MAX - RANGE_MIN) * rng.nextDouble();
	}

	public static String getRandomCompany() {
		return companies.get(rng.nextInt(companies.size()));
	}

	public static LocalDate getRandomDate() {
		return dates.get(rng.nextInt(dates.size()));
	}

	public static String getRandomOperator() {
		return operators.get(rng.nextInt(operators.size()));
	}
}

File: EbsSubscriber/EbsSubscriber/Helpers/FieldGenerator.cs
namespace EbsSubscriber.Helpers
{
    public static class FieldGenerator
    {
		private static readonly List<String> Companies = new List<string>() { "Google", "Bing", "Yahoo", "Fluentis", "Mind",
			"Amazon", "Conduent", "Continental", "Bitdefender" };
		private static readonly List<DateTime> Dates = new List<DateTime>() { new DateTime(2022, 3, 30), new DateTime(2021, 2, 11),
				new DateTime(2022, 4, 1), new DateTime(2011, 4, 20), new DateTime(2000, 1, 1) };
		private static readonly List<String> Operators = new List<string>() { "=", ">", "<", ">=", "<=", "!=" };
		public static readonly Random rng = new Random();
		public static readonly int RANGE_MIN = 90;
		public static readonly int RANGE_MAX = 99;

		public static double GetRandomDouble()
		{
			double value = RANGE_MIN + (RANGE_MAX - RANGE_MIN) * rng.NextDouble();
			return value;

			/*BigDecimal bd = BigDecimal.valueOf(value);
			bd = bd.setScale(1, RoundingMode.HALF_UP); // 1 cifra dupa virgula
			return bd.doubleValue();*/
			//		return RANGE_MIN + (RANGE_MAX - RANGE_MIN) * rng.nextDouble();
		}

		public static String GetRandomCompany()
		{
			return Companies[rng.Next(Companies.Count)];
		}

		public static DateTime GetRandomDate()
		{
			return Dates[rng.Next(Dates.Count)];
		}

		public static String GetRandomOperator()
		{
			return Operators[rng.Next(Operators.Count)];
		}
	}
}
